src/components/inputs/CopyInput/CopyInput.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from "react";
+import classnames from "classnames";
+import { CopyButton } from "../../buttons/CopyButton/CopyButton";
+import * as styles from "./CopyInput.scss";
+import { IBasicInputProps } from "../BasicInput/BasicInput";
+import { Tooltip, TooltipProps } from '../../overlays/Tooltip/Tooltip';
+
+export interface ICopyInputProps extends IBasicInputProps {
+	/* Whether value is invalid - shows the invalid state of text input if true */
+	invalid?: boolean;
+	/* Label to be shown above text input */
+	label?: string;
+	/* Message to be shown underneath an enabled input e.g. length requirement, example input, etc. */
+	message?: string;
+	/* Only show message when invalid is true - useful for only displaying error/validity CTAs */
+	onlyShowMessageWhenInvalid?: boolean;
+	/* Options for CopyButton Tooltip */
+	copyButtonTooltipOptions?: TooltipProps;
```

Comment:
Can you rename this `copyButtonTooltipProps` ? Just for clarity since that's what we'll be passing through?